packages/grant-explorer/src/features/round/ViewCartPage/SummaryContainer.tsx

```diff
@@ -0,0 +1,312 @@
+import { ChainId, PassportState } from "common";
+import { useCartStorage } from "../../../store";
+import React, { useEffect, useMemo, useState } from "react";
+import { Summary } from "./Summary";
+import ErrorModal from "../../common/ErrorModal";
+import ChainConfirmationModal from "../../common/ConfirmationModal";
+import { ChainConfirmationModalBody } from "./ChainConfirmationModalBody";
+import { ProgressStatus } from "../../api/types";
+import { modalDelayMs } from "../../../constants";
+import { useNavigate } from "react-router-dom";
+import {
+  useAccount,
+  useNetwork,
+  useWalletClient,
+  usePublicClient,
+} from "wagmi";
+import { Button } from "common/src/styles";
+import { InformationCircleIcon } from "@heroicons/react/24/solid";
+import { usePassport } from "../../api/passport";
+import useSWR from "swr";
+import { round, groupBy, uniqBy } from "lodash-es";
+import { getRoundById } from "../../api/round";
+import MRCProgressModal from "../../common/MRCProgressModal";
+import { MRCProgressModalBody } from "./MRCProgressModalBody";
+import { useCheckoutStore } from "../../../checkoutStore";
+import { parseUnits } from "viem";
+import { useConnectModal } from "@rainbow-me/rainbowkit";
+
+export function SummaryContainer() {
+  const { openConnectModal } = useConnectModal();
+  const { projects, chainToPayoutToken: payoutTokens } = useCartStorage();
+  const publicClient = usePublicClient();
+  const projectsByChain = useMemo(
+    () => groupBy(projects, "chainId"),
+    [projects]
+  );
+
+  const { data: rounds } = useSWR(projects, (projects) => {
+    const uniqueProjects = uniqBy(projects, "roundId");
+    return Promise.all(
+      uniqueProjects.map((proj) => getRoundById(proj.roundId, proj.chainId))
+    );
+  });
+
+  /** The ids of the chains that will be checked out */
+  const [chainIdsBeingCheckedOut, setChainIdsBeingCheckedOut] = useState<
+    ChainId[]
+  >(Object.keys(projectsByChain).map(Number));
+
+  /** Keep the chains to be checked out in sync with the projects in the cart */
+  useEffect(() => {
+    const chainIdsFromProjects = Object.keys(projectsByChain).map(Number);
+    if (chainIdsFromProjects.length < chainIdsBeingCheckedOut.length) {
+      setChainIdsBeingCheckedOut(chainIdsFromProjects);
+    }
+  }, [chainIdsBeingCheckedOut, projectsByChain]);
+
+  /** The ID of the current chain (from wallet) */
+  const { data: walletClient } = useWalletClient();
+
+  /** We find the round that ends last, and take its end date as the permit deadline */
+  const currentPermitDeadline =
```

Comment:
@vacekj cart page breaks because of this recent push, as roundEndTime is undefined when rounds is null. I'm pushing a fix for this now.

Reply:
This is resolved now.